File: OS4/OS4rb/os4.rb
# O of this algorithm is (!n)

# encoding: UTF-8
print ("Array lenght: ")
sizeArr =  gets.to_i
unique = 0
arr = Array.new(sizeArr)

for i in 0..(sizeArr-1)
  arr[i] = gets.to_i end
for i in 0..sizeArr
  for j in i..sizeArr
    if ((i != j) && (arr[i] == arr[j])) then
      unique += 1
      break end end end
print("Number of distinct elements: ", sizeArr - unique)

File: OS4/OS4cpp/OS4.cpp
/*O for this algorithm is (!n)*/

#include <iostream>
using namespace std;

int main()
{
	setlocale(LC_ALL, "Russian");
	int unique = 0, i, j, sizeArr;
	cout << "Array lenght: ";
	cin >> sizeArr;
	int *arr = new int[sizeArr];

	for (i = 0; i < sizeArr; i++) {
		cin >> arr[i];
	}
	for (i = 0; i != sizeArr; i++) {
		for (j = i; j != sizeArr; j++ ) {
			if ((i != j) && (arr[i] == arr[j])) {
				unique += 1;
				break;
			}
		}
	}
	cout <<"Number of distinct elements: " << (sizeArr - unique);
}
